src/OrchardCore.Modules/OrchardCore.Tenants/Views/FeatureProfiles/Create.cshtml

```diff
@@ -0,0 +1,92 @@
+            </div>
+        </div>
+        <div class="col-12">
+            <div class="form-group" asp-validation-class-for="FeatureRules">
+                <label asp-for="FeatureRules">@T["Rules"]</label>
+                <div class="form-control" style="min-height: 400px;">
+                    <div id="@Html.IdFor(x => x.FeatureRules)_editor" asp-for="FeatureRules" style="min-height: 385px;"  dir="@Orchard.CultureDir()" data-schema="@schema"></div>
+                </div>
+                <textarea asp-for="FeatureRules" hidden></textarea>
+                <span asp-validation-for="FeatureRules"></span>
+                <span class="hint">@T["The feature rules."]</span>
+            </div>
+        </div>
+    </div>
+
+    <div class="form-group">
+        <div class="btn-group">
+            <button type="submit" name="submit" class="btn btn-primary save">@T["Save"]</button>
+            <button type="button" class="btn btn-primary dropdown-toggle dropdown-toggle-split" data-reference="parent" data-toggle="dropdown" aria-haspopup="true" aria-expanded="false">
+                <span class="sr-only">@T["Toggle Dropdown"]</span>
+            </button>
+            <div class="dropdown-menu">
+                <button class="dropdown-item save-continue" name="submit" type="submit" value="SaveAndContinue">@T["and continue"]</button>
+            </div>
+        </div>
+
+        <a class="btn btn-secondary cancel" role="button" asp-route-action="Index">@T["Cancel"]</a>
+    </div>
+</form>
+
+<script at="Foot" asp-name="profile-editor" depends-on="monaco, admin">
+    $(document).ready(function () {
+        require(['vs/editor/editor.main'], function () {
+
+            var html = document.getElementsByTagName("html")[0];
+            const mutationObserver = new MutationObserver(setTheme);
+            mutationObserver.observe(html, { attributes: true });
+
+            function setTheme() {
+                var theme = html.dataset.theme;
+                if (theme === "darkmode") {
+                    monaco.editor.setTheme('vs-dark')
+                } else {
+                    monaco.editor.setTheme('vs')
+                }
+            }
+
+            setTheme();
+
+            var modelUri = monaco.Uri.parse("x://orchardcore.tenants.featureprofiles.json");
+            var editor = document.getElementById('@Html.IdFor(x => x.FeatureRules)_editor');
+            var textArea = document.getElementById('@Html.IdFor(x => x.FeatureRules)');
+            var schema = JSON.parse(editor.dataset.schema)
+            var model = monaco.editor.createModel(textArea.value, "json", modelUri);
+
+            monaco.languages.json.jsonDefaults.setDiagnosticsOptions({
+                validate: true,
+                schemas: [{
+                    uri: "x://orchardcore.tenants.featureprofiles.schema.json",
+                    fileMatch: [modelUri.toString()],
+                    schema: schema
```

Comment:
here we use a json schema generator to get auto complete, and schema validation in the monaco editor

It takes a dependency on https://www.nuget.org/packages/NJsonSchema/

`NJsonSchema` only has extra depenendencies on `NewtonSoft` and `Namotion.Reflection` 

Should be interesting to use to provide other schema's i.e. content type schemas/recipe schema's

(optional as in this case it is a pretty small schema)